### PR TITLE
feat: script to generate component

### PR DESCRIPTION
**Pull request by weblancaster**
Add script to generate component based on template.
Run npm run create:component -- NewComponentName

Closes DCOS-21786

### CODE REVIEW

File: scripts/templates/component/stories/Component.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import * as React from "react";
+import { storiesOf } from "@storybook/react";
+import { withReadme } from "storybook-readme";
+import Component from "../Component";
+
+const ComponentReadme = require("../README.md");
+
+storiesOf("Badge", module)
```
Comment (DanielMSchmidt) [inline scripts/templates/component/stories/Component.stories.tsx]: This needs to be Component then

File: scripts/create-component [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
```
Comment (DanielMSchmidt) [inline scripts/create-component]: Can we stop writing bash scripts and write this in JS? Bonus points for a small test case 👍

File: scripts/templates/component/Component.test.tsx [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+
+import Component from '.';
```
Comment (DanielMSchmidt) [inline scripts/templates/component/Component.test.tsx]: How does that work if component is not an index file?

File: scripts/templates/component/Component.test.tsx [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+
+describe('Component', () => {
+  it.skip('test something', () => {});
```
Comment (DanielMSchmidt) [inline scripts/templates/component/Component.test.tsx]: This is exactly the same as ```it('test something')```

File: scripts/templates/component/Component.test.tsx [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+
+import Component from '.';
+
+describe('Component', () => {
```
Comment (DanielMSchmidt) [inline scripts/templates/component/Component.test.tsx]: Sorry I am whining about this right now, but aren't we using double quotes in dcos-ui? Would love to have it the same

**Comment by Poltergeist** [comment]
I'll pick this up and do some changes.

File: tslint.json [hunk 6]
```diff
@@ -6,6 +6,11 @@
     "tslint-config-prettier",
     "tslint-plugin-prettier"
   ],
+  "linterOptions": {
+    "exclude": [
+      "./scripts/*"
```
Comment (DanielMSchmidt) [inline tslint.json]: Let's make this scripts/templates/* 👍

File: scripts/create-component [hunk 7]
```diff
@@ -1,13 +1,63 @@
-#!/usr/bin/env bash
+#!/usr/bin/env node
+let fs = require("fs-extra");
```
Comment (weblancaster) [inline scripts/create-component]: no big deal but should we use const on the requires?

File: tslint.json (hunk 6 above)
Comment (weblancaster) [reply tslint.json]: Especially now we will be adding js scripts.

**Pull request merged**